libraries/botbuilder/tests/teams/actionBasedMessagingExtension-fetchTask/src/adaptiveCardHelper.ts

```diff
@@ -0,0 +1,99 @@
+            version : '1.0',
+            type: 'AdaptiveCard',
+            body: [
+                {
+                    type: 'TextBlock',
+                    text: 'This is an Adaptive Card within a Task Module',
+                    weight: 'bolder',
+                },
+                { type: 'TextBlock', text: 'Enter text for Question:' },
+                { type: 'Input.Text', id: 'Question', placeholder: 'Question text here', value: userText },
+                { type: 'TextBlock', text: 'Options for Question:' },
+                { type: 'TextBlock', text: 'Is Multi-Select:' },
+                { type: 'Input.ChoiceSet', id: 'MultiSelect', value: isMultiSelect ? 'true' : 'false', style:'expanded', isMultiSelect: false,
+                    choices: [{title: 'True', value: 'true'}, {title: 'False', value: 'false'}] },
+                { type: 'Input.Text', id: 'Option1', placeholder: 'Option 1 here', value: option1 },
+                { type: 'Input.Text', id: 'Option2', placeholder: 'Option 2 here', value: option2 },
+                { type: 'Input.Text', id: 'Option3', placeholder: 'Option 3 here', value: option3 },
+            ],
+            actions: [
+                {
+                    type: 'Action.Submit',
+                    title: 'Submit',
+                    data: { 
+                         submitLocation: 'messagingExtensionFetchTask',
+                    }
+                },
+            ]
+        }
+
+        );
+
+        return <MessagingExtensionActionResponse> 
+                { task: <TaskModuleContinueResponse> { 
+                        type: 'continue', 
+                        value: <TaskModuleTaskInfo> { 
+                            title: 'Task Module Fetch Example',
+                            height: 450, 
+                            width: 500, 
+                            url: null,
+                            card: <Attachment> responseCard,
+                        }
+                    }
+                };
+    }
+    public static toAdaptiveCardAttachment(data: SubmitExampleData) : Attachment {
+        return CardFactory.adaptiveCard({
+            version: '1.0',
+            type: 'AdaptiveCard',
+            body: [
+                { type: 'TextBlock', text: 'Adaptive Card from Task Module', weight: 'bolder' },
+                { type: 'TextBlock', text: `${data.Question}`, id:"Question" },
+                { type: 'Input.Text', id: 'Answer', placeholder: 'Answer here...' },
+                { type: 'Input.ChoiceSet', id: 'Choices', isMultiSelect: Boolean(data.MultiSelect), style: 'expanded',
+                    choices: [{title: data.Option1, value: data.Option1}, 
+                            {title: data.Option2, value: data.Option2},
+                            {title: data.Option3, value: data.Option3}] },
+            ],
+            actions: [
+                { type: 'Action.Submit', title: 'Submit', data: { submitLocation: 'messagingExtensionSubmit'} },
```

Comment:
What's this `submitLocation` bit?

Reply:
Good question.  @EricDahlvang : Do you know what this is?

Reply:
The submit location, I believe, is related to where the button is clicked from. This matters because the invoke location does come through in the inbound activity. Meaning you'll know if this was kicked off from the search bar, or compose area.

Reply:
@daveta I think this information factors in to the scenario outlined [here](https://github.com/microsoft/botbuilder-js/pull/1258#issuecomment-538590166). I'm not sure what the conversation data looks like when a bot receives an invoke from the command bar

Reply:
submitLocation is not relevant... it could be removed, but is just helping demonstrate where the adaptive card submit occurs from.

Reply:
On second thought ultimately it does not, if it’s not used we should consider removing it.